### ISSUE TITLE
Getting a FOREIGN KEY constraint failed issue

### ISSUE THREAD

**Issue by CGrijalva90** [sqlitebrowser/sqlitebrowser#1638]
I have created the following tables:
```
    db.serialize(function() {
      db.run('CREATE TABLE IF NOT EXISTS `Artist` ( ' +
            '`id` INTEGER NOT NULL, ' +
            '`name` TEXT NOT NULL, ' +
            '`date_of_birth` TEXT NOT NULL, ' +
            '`biography` TEXT NOT NULL, ' +
            '`is_currently_employed` INTEGER NOT NULL DEFAULT 1, ' +
            'PRIMARY KEY(`id`) )');
    
      db.run('CREATE TABLE IF NOT EXISTS `Series` ( ' +
            '`id` INTEGER NOT NULL, ' +
            '`name` TEXT NOT NULL, ' +
            '`description` TEXT NOT NULL, ' +
            'PRIMARY KEY(`id`) )');
    
      db.run('CREATE TABLE IF NOT EXISTS `Issue` ( ' +
            '`id` INTEGER NOT NULL, ' +
            '`name` TEXT NOT NULL, ' +
            '`issue_number` INTEGER NOT NULL, ' +
            '`publication_date` TEXT NOT NULL, ' +
            '`artist_id` INTEGER NOT NULL, ' +
            '`series_id` INTEGER NOT NULL, ' +
            'PRIMARY KEY(`id`), ' +
            'FOREIGN KEY(`artist_id`) REFERENCES `Artist`(`id`), ' +
            'FOREIGN KEY(`series_id`) REFERENCES `Series`(`id`) )');
    });
```

`Artist` and `Series` table work fine and I'm able to update them and insert rows into them, but I can't seem to enter any data into the cells for the `Issues`. I'm not sure if there are some settings in it that I need to modify, because I can't even edit the cells to insert rows into them. I get the following error:

> Error adding record:
FOREIGN KEY constraint failed (INSERT INTO `Issue`(`id`,`name`,`issue_number`,`publication_date`,`artist_id`,`series_id`) VALUES (1,'',0,'',0,0);)

![screen shot 2018-11-30 at 9 20 29 pm](https://user-images.githubusercontent.com/12465672/49323196-ddc28100-f4e5-11e8-9caa-06769e5a17a1.png)

Which basically says that no information was stored in the table cells when I tried to create a new record.

I noticed that another user had a similar error but this was back in 2016 and he seemed to have fixed it himself.

**Comment by justinclift** [comment]
Interesting.  Which version is this happening with, and on what Operating System?

Asking mostly because our latest release (3.10.1) is over a year old now, and we're getting new to the next release (3.11.0).  If you're hitting this problem with version 3.10.1, it's worth having a go with the beta version we put out a few days ago.  That has heaps of fixes in it compared to 3.10.1. :smile:

**Comment by mgrojo** [comment]
@CGrijalva90 This should be solved in 3.11.0 (#1477). Can you confirm it trying with the 3.11.0 beta?

https://github.com/sqlitebrowser/sqlitebrowser/releases/tag/v3.11.0-beta1

**Comment by mgrojo** [comment]
Closing this as it's considered solved in the last release
